wandb/api.py

```diff
@@ -1138,14 +1146,22 @@ def _thread_body(self):
                 self._send(ready_chunks)
                 ready_chunks = []
 
-            if cur_time - posted_anything_time > self.HEARTBEAT_INTERVAL_SECONDS:
+            if cur_time - posted_anything_time > self.HEARTBEAT_SECONDS:
                 posted_anything_time = cur_time
-                util.request_with_retry(self._client.post,
-                                        self._endpoint, json={'complete': False, 'failed': False})
+                self._handle_response(util.request_with_retry(self._client.post,
+                                                              self._endpoint, json={'complete': False, 'failed': False}))
         # post the final close message. (item is self.Finish instance now)
         util.request_with_retry(self._client.post,
                                 self._endpoint, json={'complete': True, 'exitcode': int(finished.exitcode)})
 
+    def _handle_response(self, response):
+        """Logs dropped chunks and updates dynamic settings"""
+        if isinstance(response, Exception):
+            logging.error("dropped chunk %s" % files)
+        elif response.json().get("limits"):
```

Comment:
If response.json() fails (because a broken server sends bad json?) this line and the next line will cause the wandb process to crash. I'm not sure which exceptions response.json() can raise.